File: CS/TabsLayoutCustomization.Win/WinApplication.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Win;
using DevExpress.ExpressApp.Updating;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Xpo;

namespace TabsLayoutCustomization.Win {
   public partial class TabsLayoutCustomizationWindowsFormsApplication : WinApplication {
      public TabsLayoutCustomizationWindowsFormsApplication() {
         InitializeComponent();
         DelayedViewItemsInitialization = true;
      }

      protected override void CreateDefaultObjectSpaceProvider(CreateCustomObjectSpaceProviderEventArgs args) {
         args.ObjectSpaceProvider = new XPObjectSpaceProvider(args.ConnectionString, args.Connection);
      }
      private void TabsLayoutCustomizationWindowsFormsApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {

            e.Updater.Update();
            e.Handled = true;

      }
      private void TabsLayoutCustomizationWindowsFormsApplication_CustomizeLanguagesList(object sender, CustomizeLanguagesListEventArgs e) {
         string userLanguageName = System.Threading.Thread.CurrentThread.CurrentUICulture.Name;
         if (userLanguageName != "en-US" && e.Languages.IndexOf(userLanguageName) == -1) {
            e.Languages.Add(userLanguageName);
         }
      }
   }
}

File: VB/TabsLayoutCustomization.Win/WinApplication.vb
' Developer Express Code Central Example:
' How to Access the Document Manager
' 
' This example demonstrates how to access the Document Manager that the
' MdiShowViewStrategy uses to show Views. You will see how to locate tab captions
' to the left and orient them horizontally. For details, refer to the How to:
' Access the Document Manager
' (http://documentation.devexpress.com/#Xaf/CustomDocument3443) topic in XAF
' documentation.
' 
' You can find sample updates and versions for different programming languages here:
' http://www.devexpress.com/example=E4027

Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Win
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo

Namespace TabsLayoutCustomization.Win
   Partial Public Class TabsLayoutCustomizationWindowsFormsApplication
       Inherits WinApplication

      Public Sub New()
         InitializeComponent()
            DelayedViewItemsInitialization = True
            UseLightStyle = True
        End Sub

      Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
         args.ObjectSpaceProvider = New XPObjectSpaceProvider(args.ConnectionString, args.Connection)
      End Sub
      Private Sub TabsLayoutCustomizationWindowsFormsApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles Me.DatabaseVersionMismatch

            e.Updater.Update()
            e.Handled = True

      End Sub
      Private Sub TabsLayoutCustomizationWindowsFormsApplication_CustomizeLanguagesList(ByVal sender As Object, ByVal e As CustomizeLanguagesListEventArgs) Handles Me.CustomizeLanguagesList
         Dim userLanguageName As String = System.Threading.Thread.CurrentThread.CurrentUICulture.Name
         If userLanguageName <> "en-US" AndAlso e.Languages.IndexOf(userLanguageName) = -1 Then
            e.Languages.Add(userLanguageName)
         End If
      End Sub
   End Class
End Namespace
